cluster/gce/upgrade.sh

```diff
@@ -373,7 +373,7 @@ function do-node-upgrade() {
   # Do the actual upgrade.
   # NOTE(zmerlynn): If you are changing this gcloud command, update
   #                 test/e2e/cluster_upgrade.go to match this EXACTLY.
-  local template_name=$(get-template-name-from-version ${SANITIZED_VERSION})
+  local template_name=$(get-template-name-from-version ${SANITIZED_VERSION} ${NODE_INSTANCE_PREFIX})
```

Comment:
The scary comment above sent me down a rabbit hole and I haven't found the end of it, but I'm satisfied that this gets called by the `[Feature:NodeUpgrade]` tests

Reply:
Thanks for taking a look, I overlooked that comment. For existing uses (Linux -> NODE_INSTANCE_PREFIX) the change to get-template-name-from-version should be a no-op. I'm glad that this is covered by a test though, I'll check the upgrade tests after this gets merged to make sure I haven't broken anything.

Reply:
I've been checking some test dashboards today to make sure this PR didn't break anything in the GCE cluster upgrade path. I don't see any notable failures here:

https://testgrid.k8s.io/sig-release-master-upgrade#gce-new-master-upgrade-cluster&width=20
https://testgrid.k8s.io/sig-release-master-upgrade#gce-new-master-upgrade-cluster-new-parallel&width=20

On https://testgrid.k8s.io/sig-release-master-upgrade#gce-new-master-upgrade-cluster-new&width=20 the first test run that would have included this PR (2116) didn't go so well - the whole run timed out and a bunch of tests were skipped. I'm hoping it's not related to this PR (the same thing happened a few days ago, run 2109), I'll take another look at the results tomorrow.